File: WeLegends/andiag-mvp/src/main/java/es/coru/andiag/andiag_mvp/base/BaseFragmentPresenter.java
package es.coru.andiag.andiag_mvp.base;

import android.content.Context;
import android.support.v4.app.Fragment;

import es.coru.andiag.andiag_mvp.interfaces.BaseFragmentView;
import es.coru.andiag.andiag_mvp.interfaces.BaseView;

/**
 * Created by Canalejas on 11/12/2016.
 */

public abstract class BaseFragmentPresenter<V extends Fragment & BaseFragmentView, PV extends Context & BaseView> {

    private final static String TAG = BaseFragmentPresenter.class.getSimpleName();

    private V mView;
    private PV mParent;

    public V getView() {
        return mView;
    }

    public PV getParent() {
        return mParent;
    }

    public final void attach(V view, PV parentView) {
        this.mView = view;
        this.mParent = parentView;
        onViewAttached();
    }

    public final void detach() {
        this.mView = null;
        this.mParent = null;
        onViewDetached();
    }

    protected abstract void onViewAttached();

    protected abstract void onViewDetached();

}

File: WeLegends/app/src/main/java/es/coru/andiag/welegends/common/base/BaseFragmentPresenter.kt
package es.coru.andiag.welegends.common.base

import android.content.Context
import android.support.v4.app.Fragment

/**
 * Created by iagoc on 10/12/2016.
 */
abstract class BaseFragmentPresenter<V : Fragment, C : Context> : BasePresenter<V>() {

    var parentView: C? = null
        get private set

    fun attach(view: V, parentView: C) {
        this.parentView = parentView
        super.attach(view)
    }

    override fun detach() {
        super.detach()
        this.parentView = null
    }
}